integration/query_projection_test.go

```diff
@@ -130,3 +132,210 @@ func TestQueryProjectionElemMatch(t *testing.T) {
+			projection:    bson.D{{"value", bson.D{{"$slice", bson.A{-3, 2}}}}},
+			expectedArray: bson.A{int32(2), int32(3)},
+		},
+		"NegativeLimitSkipInt64": {
+			projection: bson.D{{"value", bson.D{{"$slice", bson.A{int64(3), -2}}}}},
+			err: &mongo.CommandError{
+				Code:    28724,
+				Name:    "Location28724",
+				Message: "First argument to $slice must be an array, but is of type: long",
+			},
+		},
+		"NegativeLimitSkipInt": {
+			projection: bson.D{{"value", bson.D{{"$slice", bson.A{3, -2}}}}},
+			err: &mongo.CommandError{
+				Code:    28724,
+				Name:    "Location28724",
+				Message: "First argument to $slice must be an array, but is of type: int",
+			},
+		},
+		"NegativeLimitSkipFloat": {
+			projection: bson.D{{"value", bson.D{{"$slice", bson.A{0.3, -2}}}}},
+			err: &mongo.CommandError{
+				Code:    28724,
+				Name:    "Location28724",
+				Message: "First argument to $slice must be an array, but is of type: double",
+			},
+		},
+		"ArgNaN": {
+			projection:    bson.D{{"value", bson.D{{"$slice", math.NaN()}}}},
+			expectedArray: bson.A{},
+		},
+		"ArgInf": {
+			projection:    bson.D{{"value", bson.D{{"$slice", math.Inf(1)}}}},
+			expectedArray: bson.A{int32(1), int32(2), int32(3), int32(4)},
+		},
+		"SingleArgNull": {
+			projection: bson.D{{"value", bson.D{{"$slice", nil}}}},
+			err: &mongo.CommandError{
+				Code: 28667,
+				Name: "Location28667",
+				Message: "Invalid $slice syntax. " +
+					"The given syntax { $slice: null } did not match the find() syntax " +
+					"because :: Location31273: $slice only supports numbers and [skip, limit] arrays :: " +
+					"The given syntax did not match the expression $slice syntax. :: caused by :: " +
+					"Expression $slice takes at least 2 arguments, and at most 3, but 1 were passed in.",
+			},
+		},
+		"NullInArr": {
+			projection: bson.D{{"value", bson.D{{"$slice", bson.A{nil}}}}},
+			err: &mongo.CommandError{
+				Code: 28667,
+				Name: "Location28667",
+				Message: "Invalid $slice syntax. The given syntax { $slice: [ null ] } " +
+					"did not match the find() syntax because :: Location31272: " +
+					"$slice array argument should be of form [skip, limit] " +
+					":: The given syntax did not match the expression $slice syntax. " +
+					":: caused by :: Expression $slice takes at least 2 arguments, " +
+					"and at most 3, but 1 were passed in.",
+			},
+		},
+		"NullInPair": {
+			projection:    bson.D{{"value", bson.D{{"$slice", bson.A{2, nil}}}}},
+			expectedArray: nil,
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			res := collection.FindOne(ctx, bson.D{}, options.FindOne().SetProjection(tc.projection))
+			err = res.Err()
+			if tc.err != nil {
+				require.Nil(t, tc.expectedArray)
+				AssertEqualError(t, *tc.err, err)
+				return
+			}
+			require.NoError(t, err)
+
+			var actual bson.D
+			err = res.Decode(&actual)
+			require.NoError(t, err)
+
+			if tc.expectedArray == nil {
+				assert.Nil(t, actual.Map()["value"])
```

Comment:
return; 
so next statement doesn't have to be in  `{}`

Reply:
done